lib/bank_accounts/savings_account.rb

```diff
@@ -0,0 +1,81 @@
+module BankAccounts
+  class SavingsAccount < Account
+    ##--------------------------------------------------------------------------
+    # CONSTANTS
+
+    # limits
+    MINIMUM_BALANCE = 10
+
+    # fees
+    WITHDRAWAL_FEE = 2.00
+
+
+    ##--------------------------------------------------------------------------
+    # INSTANCE METHODS
+
+
+    # create a new savings account.
+    def initialize(id, initial_balance)
+      if initial_balance < MINIMUM_BALANCE
+        raise ArgumentError.new("You cannot create a savings account with less than a $#{ MINIMUM_BALANCE } initial deposit.")
+      end
+
+      super(id, initial_balance)
+
+      return true
+    end
+
+
+    # add interest to account.
+    def add_interest(rate_percentage)
+      if validate_number(rate_percentage)
+        # convert rate from percentage to decimal.
+        rate_decimal = rate_percentage / 100.0
+
+        # calculate interest.
+        interest = @balance * rate_decimal
+
+        # add interest to balance.
+        update_balance(interest)
+
+        # return interest.
+        return interest
+      else
+        return false
+      end
+    end
+
+
+    # withdraw money from account.
+    def withdraw(amount)
+      if super(amount + WITHDRAWAL_FEE)
+        return @balance
+      else
+        return @balance
+      end
+    end
```

Comment:
In this specific case, I actually would ignore the return value entirely. Generally, a method should be called to produce a "side effect" or to return a value, but not both. In this project, the `withdraw` method is called for its "side effect" of updating the `@balance` instance variable -- you don't want to also have to concern yourself with what it returns. 

I would write the parent method like so:
```
def withdraw(amount)
  return unless validate_withdrawal(amount)
  if (amount <= 0)
    warn "You cannot withdraw negative funds! That sounds like a transaction that should be a deposit."
    return
  end

  update_balance(-amount)
end
```
and the method in the subclass would just be:
```
def withdraw(amount)
  super(amount + WITHDRAWAL_FEE)
end
```

More generally: I think implicit vs explicit is a matter of style to a pretty significant degree, and probably not one that you need to be very worried about right now. There's nothing necessarily wrong with being more explicit than necessary, especially right now when you have so much new stuff being crammed into your brain, and you will probably get more comfortable with Ruby's love of implicitness with time and experience. 

As for the question about checking the code to make sure of what it's returning -- there is a [whole book](http://www.confidentruby.com/) on this. Put it on your to-read list if you're interested, but I would _not_ advise reading it now -- it's very advanced and very dense (I'm working my way through it over the course of months, with multiple re-readings of every chapter).